faunadb-java/src/main/java/com/faunadb/client/query/Pagination.java

```diff
@@ -0,0 +1,177 @@
+package com.faunadb.client.query;
+
+import com.fasterxml.jackson.annotation.JsonValue;
+import com.faunadb.client.types.Value;
+import com.faunadb.client.types.Value.LongV;
+import com.google.common.base.Optional;
+import com.google.common.collect.ImmutableMap;
+
+import static java.util.Objects.requireNonNull;
+
+/**
+ * A pagination expression.
+ * <p>
+ * <b>Example:</b>
+ * We can paginate all instances of a class, with its evets, in pages with 4 elements only.
+ * <pre>{@code
+ * client.query(
+ *    Paginate(Match(Ref("indexes/all_spells"))
+ *      .withEvents(true)
+ *      .withSize(4)
+ * );
+ * }</pre>
+ *
+ * @see Language#Paginate(Expr)
+ * @see <a href="https://faunadb.com/documentation/queries#read_functions">FaunaDB Read Functions</a>
+ */
+public final class Pagination extends Expr {
```

Comment:
Just had an idea: perhaps the fluent API would read better without "with" everywhere. `withCursor` could be just `before` and `after`, too.

```java
client.query(
  Paginate(Match(Ref("indexes/all_spells")))
    .before(Ref("spells/1"))
    .size(4)
);